server/src/main/java/org/apache/druid/catalog/model/CatalogUtils.java

```diff
@@ -93,15 +100,54 @@ public static <T> T safeCast(Object value, Class<T> type, String propertyName)
     catch (ClassCastException e) {
       throw new IAE("Value [%s] is not valid for property %s, expected type %s",
           value,
-          propertyName,
+          key,
           type.getSimpleName()
       );
     }
   }
 
-  public static <T> T safeGet(Map<String, Object> map, String propertyName, Class<T> type)
+  public static <T> T safeGet(Map<String, Object> map, String key, Class<T> type)
   {
-    return safeCast(map.get(propertyName), type, propertyName);
+    return safeCast(map.get(key), type, key);
+  }
+
+  public static long getLong(Map<String, Object> map, String key)
```

Comment:
Should we catch any `ClassCastException` here as is being done in `safeGet` and return more friendly error to the user in the case of error?